### PR TITLE
enhance code sample

### PR DESCRIPTION
**Pull request by yossi-lavi**

### CODE REVIEW

File: piiano_vault_connector/src/main/java/com/demo/app/controller/MainController.java [hunk 1]
```diff
@@ -15,50 +18,59 @@ public class MainController {
...
-    @GetMapping(path="/add-user")
-    public @ResponseBody String addNewUserGet(
-            @RequestParam(value = "first_name") String firstName,
-            @RequestParam(value = "last_name") String lastName,
-            @RequestParam String username,
-            @RequestParam String password,
-            @RequestParam String email,
-            @RequestParam String country) {
-
+    @GetMapping(path="/update-user")
```
Comment (david-sackstein) [inline piiano_vault_connector/src/main/java/com/demo/app/controller/MainController.java:42]: Readability: I think you can define a POJO with a field for each RequestParam and bind the fields to the request parameters using:
http://dolszewski.com/spring/how-to-bind-requestparam-to-object/

File: piiano_vault_connector/src/main/java/com/demo/app/dal/UserRepository.java [hunk 2]
```diff
@@ -1,9 +1,14 @@
 package com.demo.app.dal;
 
+import java.util.List;
+
 import org.springframework.data.repository.CrudRepository;
 
 // This will be AUTO IMPLEMENTED by Spring into a Bean called userRepository
```
Comment (david-sackstein) [inline piiano_vault_connector/src/main/java/com/demo/app/dal/UserRepository.java:7]: nit: Remove the comment. Our customers know this.

File: piiano_vault_connector/src/main/java/com/piiano/connector/EntityTokenizer.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import static com.piiano.connector.VaultProperty.PROP;
+
+public class EntityTokenizer {
+
+    public static void tokenizeVaultProperties(Object obj, boolean onUpdate) {
+		Arrays.stream(obj.getClass().getDeclaredFields())
+				.filter(EntityTokenizer::isVaultProperty)
+				.forEach(f -> {
+					try {
+						f.setAccessible(true);
```
Comment (david-sackstein) [inline piiano_vault_connector/src/main/java/com/piiano/connector/EntityTokenizer.java:24]: Readability: Extract the lambda passed to forEach to a separate function.

**Review by david-sackstein — COMMENTED** [review]
1. The projects should use port 8123 like the default of the vault. It will take customers time to triage this and fix it. 
2. I couldn't get the tests to run. This is confusing.
3. The projects use Java 1.8. This is an old version and makes the code quite verbose (even var is not supported). Let's use 9 at least.
4. I think there should be extensive inline documentation in the code, especially in PiianoVaultConnector. This is where programmers will come for information on how to use the APIs. For instance in methods that call the EntityTokenizer and the repository, you should write "save the tokens in the vault" and then "also save the data to the MySQL repo". (It is not clear to me what is being stored where). 
5. The PiianoVaultConnector and the EntityTokenize projects are duplicated in two projects. This is confusing and makes me start comparing.
6. I think the EntityListener approach is a little naive and I would not use it if I were integrating the Vault. Let's concentrate on how to use the SDK and not how to use Java tricks which our customer is unlikely to use.
7. In the vault-connector project, which does not use  the listener, why use the EntityTokenizer that uses reflection to determine which fields to tokenize? There are two such fields. Why not just tokenize them explicitly?
8. Names are important. Why is the VaultConnecter a connector and not an SDK or at least a Client?
9. The EntityTokenizer is complex. I think some method extraction should be applied (and some inline documentation too).

**Review by david-piiano — APPROVED** [review]

**Pull request merged**